python/ray/data/dataset.py

```diff
@@ -1197,14 +1202,16 @@ def filter(
             >>> ds = ray.data.range(100)
             >>> ds.filter(lambda row: row["id"] % 2 == 0).take_all()
             [{'id': 0}, {'id': 2}, {'id': 4}, ...]
+            >> ds.filter(expr="id <= 4").take_all()
+            [{'id': 0}, {'id': 1}, {'id': 2}, {'id': 3}, {'id': 4}]
 
         Time complexity: O(dataset size / parallelism)
 
         Args:
             fn: The predicate to apply to each row, or a class type
                 that can be instantiated to create such a callable.
-            expr: An expression string that will be
-                converted to pyarrow.dataset.Expression type.
+            expr: An expression string needs to be a valid python expression that
+                will be converted to pyarrow.dataset.Expression type.
```

Comment:
```suggestion
                will be converted to ``pyarrow.dataset.Expression`` type.
```

Reply:
Optionally, I think we can remove the "will be converted to ``pyarrow.dataset.Expression`` type" part since it's an implementation detail that the user doesn't need to know about